finder-app/writer.c

```diff
@@ -0,0 +1,47 @@
+/****************************************************************************************
+@Author 	Harsh Beriwal (harsh.beriwal@colorado.edu)
+
+@course 	Advanced Embedded Software Development
+@instructor	Daniel Walkes 
+****************************************************************************************/
+
+#include <stdio.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <syslog.h> 
+#include <stdlib.h>
+#include <string.h> 
+
+int main (int argc, char *argv[])
+{
+	int fd; 
+	ssize_t wr_fl; 
+	openlog(NULL, 0 , LOG_USER);     
+	char *writefile = argv[1];					 //Storing Arguments in the Writefile and WriteStr
+	char *writestr = argv[2];
+	if(argc !=3){
+		syslog(LOG_ERR, "Invalid Number of Arguments: %d", (argc -1)); 
+		printf("Incorrect Number of Arguments\n");
+		printf("Number of Valid arguments is 2\n");
+		printf("Argument 1: Path to a File\n");
+		printf("Argument 2: String to be written within the specified file\n");
+		exit(1);    				
```

Comment:
I wouldn't exit(1) here I'd just set a `rc=-1` variable and return it at line 46.